eth2_testgen/shuffling/tgen_shuffling.py

```diff
@@ -39,24 +41,50 @@ def yaml_ValidatorStatusCode(dumper, data):
     # Config
     random.seed(int("0xEF00BEAC", 16))
     num_cases = 10
-    list_val_state = list(ValidatorStatusCode)
-    test_cases = []
 
+    test_cases = []
     for case in range(num_cases):
         seedhash = bytes(random.randint(0, 255) for byte in range(32))
-        num_val = random.randint(128, 512)
-        validators = [
-            ValidatorRecord(
-                status=random.choice(list_val_state),
-                original_index=num_val)
-            for num_val in range(num_val)
-        ]
+        idx_max = random.randint(128, 512)
+
+        validators = []
+        for idx in range(idx_max):
+            v = Validator(original_index=idx)
+            # 4/5 of all validators are active
+            if random.randint(0, 4):
```

Comment:
I think it would be a bit cleaner to use `random.random() < 0.8`, because it also allows fractions other than `(n - 1) / n`

Reply:
Also, I think this would be easier to understand without all the nesting. So I suggest to pick a random number in the beginning, and then do something like this:

```Py
if r < 0.2:
    v.activation_epoch = FAR_FUTURE_EPOCH
    v.exit_epoch = FAR_FUTURE_EPOCH
elif r < 0.2 + 0.25:
    # validator will exit in foreseeable future
elif r < 0.45 + 0.25:
    # ...
```

(even though adding the probabilities feels a bit messy, there's probably a better pattern for this)

Reply:
I really like the floating number approach. 

I'm not really sold on the `elif` with probability addition though. I don't think three-deep ifs are as scary.